### PR TITLE
First pass of battlemanager

### PR DESCRIPTION
**Pull request by RohitMoni**

### CODE REVIEW

File: Assets/BattleManager.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+public class BattleManager : ICharacterBattleManager
+{
+    private Character[] characters;
+
+    public BattleManager(Character[] _characters, BattleEnvironment environment)
+    {
+        characters = _characters;
+    }
+
+    public void DoActionToOtherCharacter(Action<Character> action)
```
Comment (atienn-unity) [inline Assets/BattleManager.cs]: Change to AttackOpponent()

File: Assets/ICharacterBattleManager.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+using System;
+
+public interface ICharacterBattleManager
+{
+    public void DoActionToOtherCharacter(Action<Character> action);
```
Comment (atienn-unity) [inline Assets/ICharacterBattleManager.cs]: Change to AttackOpponent()

File: Assets/BattleManager.cs (hunk 1 above)
Comment (RohitMoni) [reply Assets/BattleManager.cs]: Done

File: Assets/ICharacterBattleManager.cs (hunk 2 above)
Comment (RohitMoni) [reply Assets/ICharacterBattleManager.cs]: Done

**Review by atienn-unity — APPROVED** [review]

**Pull request merged**